### ISSUE TITLE
API: prevent dependent resources from being accessed directly

### ISSUE THREAD

**Issue by equill** [equill/restagraph#37]
It should not be possible to modify or delete a dependent resource as though it were a top-level resource.
Add validation to PUT and DELETE requests, to ensure that they reject any attempt to treat dependent resources this way.

GET requests should be permitted, but should return canonicalised paths instead of (unhelpful) UIDs. This aspect is covered in #77 and so is out of scope here.

**Comment by equill** [comment]
Done for PUT, POST, GET and DELETE, including updates to client tests.

Need to go back and re-check with a clear head, before merging and calling this good.

**Comment by equill** [comment]
Found and added the remaining section; it's now in `develop` and waiting on a bulk-merge to `main`.
